src/lib.rs

```diff
@@ -19,22 +19,17 @@
  */
 
 use std::path::PathBuf;
-use std::sync::Arc;
 
-use astarte_device_sdk::types::AstarteType;
-use astarte_device_sdk::{DeviceEvent, Value};
-use log::{debug, error, info, warn};
+use data::astarte_device_sdk_lib::AstarteDeviceSdkConfigOptions;
 use serde::Deserialize;
-use tokio::sync::mpsc::{channel, Receiver, Sender};
-use tokio::sync::RwLock;
-use tokio::task::JoinHandle;
+use telemetry::TelemetryInterfaceConfig;
 
-use crate::data::{Publisher, Subscriber};
 use crate::error::DeviceManagerError;
-use crate::ota::ota_handler::OtaHandler;
-use crate::telemetry::{TelemetryMessage, TelemetryPayload};
+
+pub use self::controller::Runtime;
 
 mod commands;
+pub mod controller;
```

Comment:
This can be keep private.
```suggestion
mod controller;
```